src/hook/manager.cpp

```diff
@@ -240,18 +242,20 @@ Future<DockerTaskExecutorPrepareInfo>
   // `DockerTaskExecutorPrepareInfo` can be deterministically resolved
   // (the last hook takes priority).
   vector<Future<Option<DockerTaskExecutorPrepareInfo>>> futures;
-  futures.reserve(availableHooks.size());
-
-  foreachvalue (Hook* hook, availableHooks) {
-    // Chain together each hook.
-    futures.push_back(
-        hook->slavePreLaunchDockerTaskExecutorDecorator(
-            taskInfo,
-            executorInfo,
-            containerName,
-            containerWorkDirectory,
-            mappedSandboxDirectory,
-            env));
+  synchronized (mutex) {
+    futures.reserve((*availableHooks).size());
```

Comment:
How about `availableHooks->size()` here and all other `(*availableHooks).method()` cases in this patch? That would be more idiomatic with our code base

Reply:
Agree with it.